DeadCrumbs/app/src/main/java/dead/crumbs/ui/DeadReckoningService.kt

```diff
@@ -0,0 +1,73 @@
+package dead.crumbs.ui
+
+import android.content.Context
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
+import android.util.Log
+import android.app.Service
+import android.content.Intent
+import android.os.Binder
+import android.os.IBinder
+
+class DeadReckoningService : Service(), SensorEventListener{
+
+    private lateinit var sensorManager: SensorManager
+    private var stepCounterSensor: Sensor? = null
+
+    override fun onCreate() {
+        super.onCreate()
+
+        sensorManager = getSystemService(Context.SENSOR_SERVICE) as SensorManager
+        stepCounterSensor = sensorManager.getDefaultSensor(Sensor.TYPE_STEP_COUNTER)
+
+        sensorManager.registerListener(this, stepCounterSensor, SensorManager.SENSOR_DELAY_NORMAL)
```

Comment:
There is in general quite a large delay before the onSensorChanged is called. I have a hard time detecting <10 steps and getting it outputted into the log.  Most times the first output (other than the 0 output) in log is >10 steps. I don't think that it is an issue in your code but rather due to the Step Counter sensor filtering out false positives. This may be a big issue later in the development of DeadReckoning.

I tried to change SENSOR_DELAY_NORMAL -> SENSOR_DELAY_FASTEST and it may improve upon this slightly.

Reply:
I'm not sure what's best right now for SENSOR_DELAY, but I added a todo to consider it later